frontend/src/metabase/admin/settings/selectors.js

```diff
@@ -608,6 +595,15 @@ export const getSettings = createSelector(
     ),
 );
 
+/**
+ * @typedef {import("metabase-types/store").State} State
+ *
+ * @type {(state: State) => Record<keyof import("metabase/selectors/settings").GetSettings<State>, any>}
+ */
+export const getSettingsByKey = createSelector(getSettings, settings =>
```

Comment:
Replace all places that uses `getSettings` with `getSettingsByKey` to simplify logic.